Code/GraphMol/Depictor/Templates.h

```diff
@@ -0,0 +1,78 @@
+//
+//  Copyright (C) 2003-2022 Greg Landrum and other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+
+#include <GraphMol/ROMol.h>
+#include <GraphMol/SmilesParse/SmilesParse.h>
+#include <GraphMol/MolOps.h>
+
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+
+class CoordinateTemplates {
+ public:
+  //! returns a reference to the singleton CoordinateTemplates
+  /*
+      \return a reference to the singleton CoordinateTemplates
+
+      <b>Notes:</b>
+        - if the singleton CoordinateTemplates has already been instantiated
+          the singleton will be returned, otherwise the singleton will
+          be constructed.
+
+   */
+  static CoordinateTemplates& getRingSystemTemplates() {
+    static CoordinateTemplates template_mols;
+    return template_mols;
+  }
+
+  bool hasTemplateOfSize(unsigned int atom_count) {
+    if (m_templates.find(atom_count) != m_templates.end()) {
+        return true;
+    }
+    return false;
+  }
+
+  const std::vector<RDKit::ROMol*>& getMatchingTemplates(unsigned int atom_count) {
+    return m_templates[atom_count];
+  }
+
+ private:
+  CoordinateTemplates() {
+    // load templates into m_templates map by atom count
+    std::string rdbase = getenv("RDBASE");
+    std::string fpath = rdbase + "/Code/GraphMol/Depictor/templates.smi";
```

Comment:
This is fine during development and testing, but before we merge this, the template file needs to either move to somewhere in `$RDBASE/Data` or we need to actually incorporate its contents as code. I'd lean towards direct incorporation, but I'm definitely open to discussion.

Reply:
I moved this to `$RDBASE/Data` for now but I can move it to the templates header if you'd prefer that. My reservation is that I envision there eventually being a couple hundred templates along with around 50 macrocycle templates, and I could see that being a little cluttered. I also use the templates file to test that all the templates are working correctly (see `testUseRingSystemTemplates` in `testDepictor.py`), so I think it'd be useful to have a file with all these templates either way.